Common/Ubuntu/ap_camera_non_gps/feature_match.py

```diff
@@ -0,0 +1,182 @@
+import numpy as np
+import cv2 as cv
+from matplotlib import pyplot as plt
+import subprocess
+import pyexiv2, json
+import math
+
+x_offset = 0
+y_offset = 0
+
+MIN_MATCH_COUNT = 30
+
+LOCATION_SCALING_FACTOR_INV = 89.83204953368922
+
+inCMX = 0
+inCMY = 0
+
+focal = 476.7030836014194
+
+
+
+def readExif(image):
+    with open(image, 'rb') as f:
+        with pyexiv2.ImageData(f.read()) as img:
+            data = img.read_exif()
+            print(data['Exif.Photo.UserComment'])
+    f.close()
+
+def modifyExif(image,data=None):
+    with open(image, 'rb+') as f:
+        with pyexiv2.ImageData(f.read()) as img:
+            matadata = data
+            dict1 = {'Exif.Photo.UserComment': json.dumps(metadata)}
+            img.modify_exif(dict1)
+            
+            f.seek(0)
+            f.truncate()
+            f.write(img.get_bytes())
+        f.seek(0)
+        with pyexiv2.ImageData(f.read()) as img:
+            result = img.read_exif()
+            #print(result)
+    f.close()
+
+
+def compParam(img1,img2):
+    global inCMY, inCMX
+    global x_offset, y_offset
+    img1 = cv.imread(img1,0)
+    img2 = cv.imread(img2,0)
+    #img2 = cv.resize(img2, (1778,866))
+    sift = cv.SIFT_create(nfeatures = 10000)
+
+    kp1, des1 = sift.detectAndCompute(img1,None)
+    kp2, des2 = sift.detectAndCompute(img2,None)
+    FLANN_INDEX_KDTREE = 1
+    index_params = dict(algorithm = FLANN_INDEX_KDTREE, trees = 5)
+    search_params = dict(checks = 50)
```

Comment:
If you plan to run this to run at a regular rate, it might be wise to iteratively adjust the checks over time to optimize the computation time against the desired output rate. It might depend if you want to run this algorithm on every frame, or skip frames in order to get higher checks.

Reply:
Understood sir! You mean it could be better to do frequent checks on image frames rather than continuous checks on frames, I'll make it to check for features at frequent intervals